### PR TITLE
feat: add markdown check in check target

### LINKED ISSUE #102: 🛠️ [TASK] : Convert Markdown to standard `check` targets.
### Summary

Make Markdown conform to `check` target pattern

### Description

Currently Markdown  checks are independent of our standard CI pipeline.

We still need the ability to efficiently run it locally, and to run `fix` for Markdown which also needs local support.

But in CI we can use a standard `check` target for these and then remove the github action which is triggering these.

This will help better align our CI pipeline and reduce the number of github actions we need to enforce.

We would also need to add Markdown guides, which would be a part of this ticket.
This ticket also includes updating both:
https://github.com/input-output-hk/catalyst-voices
https://github.com/input-output-hk/hermes

To use the update check target, each will need a new issue raised to track this issue, and for the PR in those repos to be raised against that Issue.

### PR DESCRIPTION
**Pull request by bkioshn**
# Description
Adding markdown check to standard `check` target.

## Description of Changes
- Adding markdown check to `check` target to be able to perform a remote standard checking. The changed includes`check-markdown` and `CHECK`
- Update the `davidanson/markdownlint-cli2-rules` from v0.10.0 to v0.11.0
- Rename the rules for markdownlint-disable
- Add readme `/docs/src/guides/markdown.md`
- Fix readme `/earthly/mdlint/Readme.md`

## Please confirm the following checks

* [x] My code follows the style guidelines of this project
* [x] I have performed a self-review of my code
* [x] I have commented my code, particularly in hard-to-understand areas
* [x] I have made corresponding changes to the documentation
* [x] My changes generate no new warnings
* [x] I have added tests that prove my fix is effective or that my feature works
* [x] New and existing unit tests pass locally with my changes
* [x] Any dependent changes have been merged and published in downstream module

### CODE REVIEW

File: docs/src/guides/markdown.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+<!-- markdownlint-enable max-one-sentence-per-line -->
+
+### Remotely
+
+#### Running check
+
+Running `check` stage can be done by referencing to the `check` target created in this repository by adding
+
+``` Earthfile
+IMPORT github.com/input-output-hk/catalyst-ci/:<tag>
```
[the remark this answers is not in the document]
Comment (bkioshn) [reply docs/src/guides/markdown.md:47]: can we call the `check` target not the function?

[the remark this answers is not in the document]
Comment (bkioshn) [reply docs/src/guides/markdown.md]: move to `/earthly/mdlint/Readme.md`

File: Earthfile [hunk 2]
```diff
@@ -4,14 +4,18 @@ FROM debian:stable-slim
 
 # cspell: words livedocs sitedocs
 
+# Check Markdown in this repo.
```
[the remark this answers is not in the document]
Comment (Mr-Leshiy) [reply Earthfile:7]: No, it is not important, we don't have an Earthly docs generation based on our source code and comments in it like it works with Rust docs or any other automatic docs generation.

**Comment by Mr-Leshiy** [comment]
look at this comment: https://github.com/input-output-hk/catalyst-ci/pull/118#issuecomment-1840422522

**Review by stevenj — APPROVED** [review]
LGTM

**Pull request merged**